portfolio/src/test/java/com/google/sps/CalendarGroupTest.java

```diff
@@ -0,0 +1,318 @@
+package com.google.sps;
+
+import com.google.sps.data.CalendarEvent;
+import com.google.sps.data.CalendarGroup;
+import com.google.sps.data.TimeRange;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CalendarGroupTest {
+
+  /** Start time of the scheduling block must not be later then end time. */
+  @Test(expected = IllegalArgumentException.class)
+  public void startTimeAfterEndTime() {
+    Instant startTime = Instant.now();
+    Instant endTimeBefore = startTime.minusSeconds(1000);
+    Instant endTimeAfter = startTime.plusSeconds(1000);
+
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+    CalendarEvent event = new CalendarEvent("event", startTime, endTimeAfter);
+    events.add(event);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTimeBefore);
+  }
+
+  /** Tests for deleting a time range that doesn't exsit in the list of all free time ranges. */
+  @Test(expected = IllegalArgumentException.class)
+  public void deleteTimeRangeNonExistent() {
+    // Events:       |--One--|   |--Two--|
+    // Possible: |---------------------------|
+    // Free:     |---|       |---|       |---|
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+
+    Instant eventOneStart = Instant.now();
+    Instant eventOneEnd = eventOneStart.plusSeconds(1000);
+    Instant eventTwoStart = eventOneEnd.plusSeconds(1000);
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(1000);
+
+    Instant startTime = eventOneStart.minusSeconds(1000);
+    Instant endTime = eventTwoEnd.plusSeconds(1000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+
+    Instant wrongStart = eventOneStart.minusSeconds(50);
+    Instant wrongEnd = eventOneStart;
+
+    eventGroup.deleteFreeTimeRange(TimeRange.fromStartEnd(wrongStart, wrongEnd));
+  }
+
+  /** Tests for correct free times with non-overlapping events. */
+  @Test
+  public void nonoverlappingEvents() {
```

Comment:
This tests the calculateFreeTimeRanges() method more specifically so maybe the name should reflect that, possibly "freeTimeRangesNonOverlappingEvents()".

Reply:
Updated.